rmw_opendds_cpp/src/rmw_client.cpp

```diff
@@ -27,309 +27,233 @@
 #include "process_topic_and_service_names.hpp"
 #include "type_support_common.hpp"
 
+#include <dds/DCPS/Marked_Default_Qos.h>
+
 // Uncomment this to get extra console output about discovery.
 // This affects code in this file, but there is a similar variable in:
 //   rmw_opendds_shared_cpp/shared_functions.cpp
 // #define DISCOVERY_DEBUG_LOGGING 1
 
 extern "C"
 {
+rmw_ret_t clean_client(OpenDDSNodeInfo & node_info, rmw_client_t * client)
```

Comment:
The purpose of clean_client() is to clear (if client is unnull) client's members and then free the client, so the client is passed as a point. In order to do so, node needs to do some housekeeping: remove_information and trigger_graph_guard_condition. That's why node is passed as a mutable reference, not a const reference, nor a point (since node is not freed here).